open-machine-learning-jupyter-book/ml-fundamentals/Build a Web App to use a Mchine Learning model.md

```diff
@@ -0,0 +1,352 @@
+        </form>
+
+        <p>{{ prediction_text }}</p>
+
+        </div>
+
+    </div>
+
+    </body>
+</html>
+```
+
+```{note}
+Take a look at the templating in this file. Notice the 'mustache' syntax around variables that will be provided by the app, like the prediction text: `{{}}`. There's also a form that posts a prediction to the `/predict` route.
+
+Finally, you're ready to build the python file that drives the consumption of the model and the display of predictions:
+```
+
+In `app.py` add:
+
+```python
+import numpy as np
+from flask import Flask, request, render_template
+import pickle
+
+app = Flask(__name__)
+
+model = pickle.load(open("./ufo-model.pkl", "rb"))
+
+
+@app.route("/")
+def home():
+    return render_template("index.html")
+
+
+@app.route("/predict", methods=["POST"])
+def predict():
+
+    int_features = [int(x) for x in request.form.values()]
+    final_features = [np.array(int_features)]
+    prediction = model.predict(final_features)
+
+    output = prediction[0]
+
+    countries = ["Australia", "Canada", "Germany", "UK", "US"]
+
+    return render_template(
+        "index.html", prediction_text="Likely country: {}".format(countries[output])
+    )
+
+
+if __name__ == "__main__":
+    app.run(debug=True)
+```
+
+```{note}
+when you add [`debug=True`](https://www.askpython.com/python-modules/flask/flask-debug-mode) while running the web app using Flask, any changes you make to your application will be reflected immediately without the need to restart the server. Beware! Don't enable this mode in a production app.
+```
+
+If you run `python app.py` or `python3 app.py` - your web server starts up, locally, and you can fill out a short form to get an answer to your burning question about where UFOs have been sighted!
+
+Before doing that, take a look at the parts of `app.py`:
+
+1. First, dependencies are loaded and the app starts.
+2. Then, the model is imported.
+3. Then, index.html is rendered on the home route.
+
+On the `/predict` route, several things happen when the form is posted:
+
+1. The form variables are gathered and converted to a numpy array. They are then sent to the model and a prediction is returned.
+2. The Countries that we want displayed are re-rendered as readable text from their predicted country code, and that value is sent back to index.html to be rendered in the template.
+
+Using a model this way, with Flask and a pickled model, is relatively straightforward. The hardest thing is to understand what shape the data is that must be sent to the model to get a prediction. That all depends on how the model was trained. This one has three data points to be input in order to get a prediction.
+
+In a professional setting, you can see how good communication is necessary between the folks who train the model and those who consume it in a web or mobile app. In our case, it's only one person, you!
+
+---
+
+## Your trun! 🚀 
```

Comment:
You also need to add the assignment here, https://github.com/microsoft/ML-For-Beginners/blob/main/3-Web-App/1-Web-App/README.md#assignment